core/services/gateway/gateway.go

```diff
@@ -183,25 +195,30 @@ func (g *gateway) ProcessRequest(ctx context.Context, rawRequest []byte) (rawRes
 	return rawResponse, api.ToHttpErrorCode(api.NoError)
 }
 
-func (g *gateway) ProcessServiceRequest(ctx context.Context, request *jsonrpc.Request) (rawResponse []byte, httpStatusCode int) {
-	service, ok := g.services[request.ServiceName()]
+func (g *gateway) processJsonRpc2Request(ctx context.Context, request jsonrpc2.Request, jwtToken string) (rawResponse []byte, httpStatusCode int) {
+	handlerType := request.ServiceName()
+	h, ok := g.handlers[handlerType]
```

Comment:
hmm not sure if this would work. It seems that `g.handlers` is a map with donID as keys, not handlerType